### PR TITLE
[POAE7-2786] BatchProcessor interfaces to support pipeline with Cross…

### PR DESCRIPTION
**Pull request by ZJie1**
…Join

### What changes were proposed in this pull request?

BatchProcessor interfaces to support pipeline with CrossJoin 

### Why are the changes needed?

Add CrossJoin Support.

### Does this PR introduce _any_ user-facing change?

No
### How was this patch tested?

No, since there still have some feature gaps in v2s and s2eu

### CODE REVIEW

**Comment by ZJie1** [comment]
@chaojun-zhang @qiuyangshen Please help take a review.

File: cpp/src/cider/exec/processor/DefaultBatchProcessor.cpp [hunk 1]
```diff
@@ -128,6 +131,12 @@ void DefaultBatchProcessor::feedHashBuildTable(
   this->codegen_context_->setHashTable(hashTable.get());
 }
 
+void DefaultBatchProcessor::feedCrossBuildData(Batch crossData) {
```
Comment (qiuyangshen) [inline cpp/src/cider/exec/processor/DefaultBatchProcessor.cpp:134]: May be use const std::shared_ptr<Batch>& crossData will be better

File: cpp/src/cider/include/cider/processor/BatchProcessorContext.h [hunk 2]
```diff
@@ -52,9 +59,19 @@ class BatchProcessorContext {
...
+  }
+
+  const CrossJoinColumnBatchSupplier& getCrossJoinColumnBatchSupplier() const {
+    return columnBatchSupplier_;
+  }
+
  private:
   std::shared_ptr<CiderAllocator> allocator_;
   HashBuildTableSupplier buildTableSupplier_;
+  CrossJoinColumnBatchSupplier columnBatchSupplier_;
```
Comment (chaojun-zhang) [inline cpp/src/cider/include/cider/processor/BatchProcessorContext.h:74]: maybe rename to 'crossBuildTableSupplier' and rename 'buildTableSupplier' to 'hashBuildTableSupplier' sounds better?

File: cpp/src/cider/include/cider/processor/BatchProcessorContext.h [hunk 3]
```diff
@@ -52,9 +59,19 @@ class BatchProcessorContext {
     return buildTableSupplier_;
   }
 
+  void setCrossJoinColumnBatchSupplier(
```
Comment (chaojun-zhang) [inline cpp/src/cider/include/cider/processor/BatchProcessorContext.h:62]: setCrossJoinBuildTableSupplier sounds better ?

**Review by chaojun-zhang — APPROVED** [review]
LGTM

**Review by qiuyangshen — APPROVED** [review]

**Review by qiuyangshen — APPROVED** [review]

**Review by winningsix — APPROVED** [review]

**Pull request merged**